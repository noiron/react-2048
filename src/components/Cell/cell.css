.cell {
    width: 1rem;
    height: 1rem;
    box-sizing: border-box;
    border-radius: 5px;
    margin-right: .06rem;
    background-color: #303030;
}

.inner {
    width: 90%;
    height: 90%;
    font-size: .32rem;
    line-height: 3;
    text-align: center;
    color: #000;
    text-shadow: 1px 1px 2px #ddd;
}


/* Colors from http://materialuicolors.co/  Orange */
/* .color-2 {
    background-color: #FFF3E0;
}

.color-4 {
    background-color: #FFE0B2;
}

.color-8 {
    background-color: #FFCC80;
}

.color-16 {
    background-color: #FFB74D;
}

.color-32 {
    background-color: #FFA726;
}

.color-64 {
    background-color: #FF9800;
}

.color-128 {
    background-color: #FB8C00;
}

.color-256 {
    background-color: #F57C00;
}

.color-512 {
    background-color: #EF6C00;
}

.color-1024 {
    background-color: #E65100;
}

.color-2048 {
    background-color: #FF9100;
}

.color-4096 {
    background-color: #FF6D00;
}

.color-8192 {
    background-color: #FFD180;
} */

.color-2 {
    background-color: #A8D8EA;
}

.color-4 {
    background-color: #AA96DA;
}

.color-8 {
    background-color: #FCBAD3;
}

.color-16 {
    background-color: #8BC34A;
}

.color-32 {
    background-color: #F38181;
}

.color-64 {
    background-color: #FCE38A;
}

.color-128 {
    background-color: #9E9E9E;
}

.color-256 {
    background-color: #EAFFD0;
}

.color-512 {
    background-color: #95E1D3;
}

.color-1024 {
    background-color: #03A9F4;
}

.color-2048 {
    background-color: #00ADB5;
}

.color-4096 {
    background-color: #393E46;
}

.color-8192 {
    background-color: #3490DE;
}
