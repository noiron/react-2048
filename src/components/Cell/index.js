import React from 'react';
import PropTypes from 'prop-types';
import styles from './cell.css';


export default class Cell extends React.PureComponent {
    static propTypes = {
        value: PropTypes.number.isRequired
    };

    render() {
        const { value } = this.props;

        return (
            <div className={[styles.cell, styles[`color-${value}`]].join(' ')}>
                <div className={styles.inner}>{value || null}</div>
            </div>
        )
    }
}