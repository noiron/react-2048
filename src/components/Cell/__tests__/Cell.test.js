import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Cell from '..';

Enzyme.configure({ adapter: new Adapter() });

describe('<Cell />', () => {
    it('component render', () => {
        let cell = renderer.create(<Cell value={32} />).toJSON();
        expect(cell).toMatchSnapshot();
    })
})
