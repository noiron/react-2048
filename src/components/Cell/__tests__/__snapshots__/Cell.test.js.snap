// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> component render 1`] = `
<div
  className=" "
>
  <div
    className={undefined}
  >
    32
  </div>
</div>
`;
